test/extended/builds/long_names.go

```diff
@@ -0,0 +1,90 @@
+package builds
+
+import (
+	"path/filepath"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[builds][Slow] extremely long build/bc names are not problematic", func() {
```

Comment:
Next time around, consider doing an integration test by creating mock objects (mock oc client object that returns builds/buildconfigs and records what delete operations are called) for the reaper.  Then you can just invoke reaper.Stop() directly w/o having all the cluster infrastructure, running builds, etc.

(The problem w/ extended tests are that they are slower, and because they are slower, most of them do not get run as part of the merge queue, which makes it easier for people to accidentally regression functionality until we catch it when running the extended tests explicitly)